library/src/main/kotlin/com/chuckerteam/chucker/api/ChuckerInterceptor.kt

```diff
@@ -68,25 +71,27 @@ public class ChuckerInterceptor private constructor(
     override fun intercept(chain: Interceptor.Chain): Response {
         val transaction = HttpTransaction()
         val request = chain.request()
-
-        requestProcessor.process(request, transaction)
-
-        val response = try {
-            chain.proceed(request)
-        } catch (e: IOException) {
-            transaction.error = e.toString()
-            collector.onResponseReceived(transaction)
-            throw e
+        return if(skipPaths.any { it == request.url.encodedPath }) {
+             chain.proceed(request)
+        } else {
```

Comment:
you won't be catching `IOException` here.
Please update this logic so that:

1. You check if the path has to be skipped it, and save it in a boolean variable.
2. If true, call `requestProcessor.process`, `collector.onResponseReceived` (for the error)` and `responseProcessor.process`

Reply:
A little confused about this one.

Did you mean something like the following:-
```
@Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val transaction = HttpTransaction()
        val request = chain.request()
        val skipProcessing = skipPaths.any { it == request.url.encodedPath } 
        return if(skipProcessing) {
             chain.proceed(request)
        } else {
            requestProcessor.process(request, transaction)
            val response = proceedRequest(chain, request, transaction)
            responseProcessor.process(response, transaction)
        }
    }

    private fun proceedRequest(
        chain: Interceptor.Chain,
        request: Request,
        transaction: HttpTransaction
    ): Response {
        val response = try {
            chain.proceed(request)
        } catch (e: IOException) {
            transaction.error = e.toString()
            collector.onResponseReceived(transaction)
            throw e
        }
        return response
    }

Reply:
I meant something like:

```kotlin
		val shouldProcess = !(skipPaths.any { it == request.url.encodedPath })
		if (shouldProcess) {
		    requestProcessor.process(request, transaction)
		}

        val response = try {
            chain.proceed(request)
        } catch (e: IOException) {
            transaction.error = e.toString()
			if (shouldProcess) {
                collector.onResponseReceived(transaction)
            }
            throw e
        }

        return if (shouldProcess) responseProcessor.process(response, transaction) else response
```

Reply:
Ah! I see what you meant. This probably saves us from the redundant callsite of  `chain.proceed(..)` as well.

Reply:
Awesome I'll check this out later today 👍